packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -206,6 +206,8 @@ Future<XcodeBuildResult> buildXcodeProject({
   bool codesign: true,
   bool usesTerminalUi: true,
 }) async {
+  upgradePbxProjWithFlutterAssets(app.name);
```

Comment:
You'll need an `await` here, otherwise it might not be done by the time the build runs.